src/chrome/content/scaffold/scaffold.js

```diff
@@ -303,6 +303,51 @@ var Scaffold = new function() {
 	});
 
 	/*
+	 * add template code
+	 */
+	this.addTemplate = Zotero.Promise.coroutine(function* (template, second) {
```

Comment:
We can now use `async function`/`await` instead of `Zotero.Promise.coroutine(function*`/`yield`

Reply:
I will leave this for you to do afterwards, as there are more of this function calls, and I am not sure how to change that exactly.